packages/cli-repl/src/cli-repl.ts

```diff
@@ -266,7 +266,11 @@ export class CliRepl implements MongoshIOProvider {
         this.warnAboutInaccessibleFile(err, path),
     });
 
-    await this.logManager.cleanupOldLogFiles();
+    // Do not wait for log cleanup and log errors if MongoLogManager throws any.
+    void this.logManager.cleanupOldLogFiles().catch((err) => {
+      this.bus.emit('mongosh:error', err, 'log');
+    });
+
     markTime(TimingCategories.Logging, 'cleaned up log files');
```

Comment:
Actually, this line would need to go into a `.finally()` for the Promise above

Reply:
Good catch!